src/main/java/pro/sky/skyprospringtyulkovcw2/service/Impl/QuestionServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package pro.sky.skyprospringtyulkovcw2.service.Impl;
+
+import org.springframework.stereotype.Service;
+import pro.sky.skyprospringtyulkovcw2.exeption.QuestionAlreadyAddedExeption;
+import pro.sky.skyprospringtyulkovcw2.exeption.NoQuestionExeption;
+import pro.sky.skyprospringtyulkovcw2.exeption.QuestionNotFoundExeption;
+import pro.sky.skyprospringtyulkovcw2.model.Question;
+import pro.sky.skyprospringtyulkovcw2.service.QuestionService;
+
+import java.util.*;
+import java.util.concurrent.ThreadLocalRandom;
+
+@Service
+public class QuestionServiceImpl implements QuestionService {
+
+    private final Set<Question> questions = new HashSet<>();
+
+    @Override
+    public Question add(String question, String answer) {
+        Question q = new Question(question, answer);
+        if (!questions.contains(q)) {
+            throw new QuestionAlreadyAddedExeption();
+        }
+        questions.add(q);
+        return q;
```

Comment:
дублирование кода, лучше сделать делегацию:
```suggestion
        return add(new Question(question, answer));
```